### ISSUE TITLE
Representative sample of some sort.

### ISSUE THREAD

**Issue by JakeWharton** [JakeWharton/RxBinding#92]
* Use Toolbar, react to action item presses.
 * Use SwipeRefreshLayout, react to pull-to-refresh
 * Use EditText, react to text typing
 * Use TextViews, bind text to a composite stream

This, or some handful of our operators, should be combined in a representative sample to showcase the ease of binding and composition afforded by these APIs to solve complex interaction problems.

**Comment by hzsweers** [comment]
A good example for edittext is username inputs when creating/editing an account. Take the text changes and pipe them right into a nice rx chain to validate and check availability. Loose mobile psuedocode below

> watch for text changes
> debounce/throttle so they have time to finish typing
> filter out invalid usernames
> flatmap network request to check username availability
> observer's onError and onNext indicate the result to the user

We do it in our app, and it's always one of my go-to examples whenever I show someone how we use rxjava. A similar example could be auto complete filling where the text is first sent to the server and items are returned.

**Comment by JakeWharton** [comment]
Not sure this is still useful because it doesn't have a strong scope. Maybe we just list a few open source apps which are using this library? I also really don't like that approach since people will make PRs wanting to add every and any app which might not actually be using this library well.

**Comment by hzsweers** [comment]
I think linking some examples would be a good middle ground.

**Comment by JakeWharton** [comment]
Yep. I don't know any other than @f2prateek's Bee. Anyone know more?

**Comment by hzsweers** [comment]
I could put together an example of the classic debounced search query in a gist. I've done it enough times in apps that I should probably just pull it out.

Are you looking for full app examples or also gists and articles? The only concern with external examples though is that they can/will eventually get out of date.

**Comment by feresr** [comment]
It would be great to have basic sample on how to integrate RxBindings with common architectures (MVP, MVVM) or how to use with the new arch components

**Comment by JakeWharton** [comment]
That sample would be the same as if you use normal listeners or
Observable.create yourself to wrap callbacks though. Seems vastly out of
scope.

**Comment by RobLewis** [comment]
I'm fairly lost as to how to even start thinking about this. Need a couple of examples to crack open the wall of ignorance. (I'm familiar with RxJava but not in this context.)
